shared/middleware/files_api.rb

```diff
@@ -465,23 +465,23 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
     end
   end
 
+  #
+  # NOTE: The files API that we expose is case-insensitive, though AWS s3 is case-sensitive. As a
+  # result, we normalize s3 filenames to be downcased. That said, we maintain a case-sensitive
+  # manifest.
+  #
   def files_put_file(encrypted_channel_id, filename, body)
-    bad_request if filename.downcase == FileBucket::MANIFEST_FILENAME
+    downcased_filename = filename.downcase
+    bad_request if downcased_filename == FileBucket::MANIFEST_FILENAME
```

Comment:
:( Looks like coverage isn't getting reported correctly for shared tests?  Maybe I'll take a look today.